### PR TITLE
Fixed :x: for quotes, other small fixes/cleanups

### LINKED ISSUE #322: Message links from other guilds cause nullref exception
https://paste.mod.gg/posavohupu.json

### PR DESCRIPTION
**Pull request by calledude**
* Adds the ability to remove a quote with the :x: reaction.
* The quote now also contains a jump-link to the quoted message itself.
* It no longer freaks out when someone links a message that the bot cannot see. (i.e. a message from another guild, or a channel that it doesn't have access to.)

The end result can be seen below.

### From this: 

discord_2019-03-03_18-51-03

### To this:
bild

Fixes #322

### CODE REVIEW

File: Modix.Services/Quote/MessageLinkBehavior.cs [hunk 1]
```diff
@@ -70,33 +79,29 @@ private async Task OnMessageReceivedAsync(SocketMessage message)
...
-                if (msg == null) { return; }
+                var embed = quoteService.BuildQuoteEmbed(message, quoter);
+                if (embed == null) return;
 
-                await SelfExecuteRequest<IQuoteService>(async quoteService =>
-                {
-                    var embed = quoteService.BuildQuoteEmbed(msg, quoter)?.Build();
-                    if (embed == null) { return; }
+                embed.Fields.First(d => d.Name == "Quoted by")
+                .Value += $" from **[#{message.Channel.Name}]({message.GetJumpUrl()})**";
```
Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: Should this just be part of the `BuildQuoteEmbed()` logic?

File: Modix.Services/Quote/MessageLinkBehavior.cs [hunk 2]
```diff
@@ -70,33 +79,29 @@ private async Task OnMessageReceivedAsync(SocketMessage message)
...
+                embed.WithTimestamp(message.Timestamp);
+
+                var removableMessage = await targetChannel.SendMessageAsync(string.Empty, embed: embed.Build());
+                await SelfExecuteRequest<IAutoRemoveMessageService>(async messageRemoveService
+                    => await messageRemoveService.RegisterRemovableMessageAsync(removableMessage, quoter));
+            });
         }
 
-        private async Task<bool> IsQuote(ulong channelId, ulong messageId)
+        private async Task<bool> IsQuote(IMessage message)
```
Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: Can this just be a synchronous method now?

Comment (calledude) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Yup!

File: Modix.Services/Quote/MessageLinkBehavior.cs (hunk 1 above)
Comment (calledude) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Yep.

Comment (calledude) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: I'll change that in this PR and take a closer look at the entire `QuoteService` later on today, just a quick glance on it and saw a few improvements that could be made.

**Pull request merged**